adapters/http-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -253,42 +256,48 @@ protected void onStartupSuccess() {
     /**
      * Creates the router for handling requests.
      * <p>
-     * This method creates a router instance along with a route matching all request. That route is initialized with the
-     * following handlers and failure handlers:
+     * This method creates a router instance along with a route matching all requests. The handler set
+     * on that route makes sure that
      * <ol>
-     * <li>a default failure handler,</li>
-     * <li>a handler to keep track of the tracing span created for the request by means of the Vert.x/Quarkus
-     * instrumentation,</li>
-     * <li>a handler to add a Micrometer {@code Timer.Sample} to the routing context,</li>
-     * <li>a handler to log when the connection is closed prematurely,</li>
+     * <li>the tracing span created for the request by means of the Vert.x/Quarkus
+     * instrumentation is available via {@link HttpServerSpanHelper#serverSpan(RoutingContext)},</li>
+     * <li>a Micrometer {@code Timer.Sample} is added to the routing context,</li>
+     * <li>there is log output when the connection is closed prematurely.</li>
      * </ol>
+     * Also a default failure handler is set.
      *
      * @return The newly created router (never {@code null}).
      */
     protected Router createRouter() {
 
-        // route (failure) handlers are added in a specific order here!
+        final var customTags = getCustomTags();
+
         final Router router = Router.router(vertx);
+        final DefaultFailureHandler defaultFailureHandler = new DefaultFailureHandler();
         final Route matchAllFailuresRoute = router.route();
-        // failure handler is added on a separate route for which no name is set, otherwise all tracing spans
-        // of failed HTTP requests would get that route name as span name
-        matchAllFailuresRoute.failureHandler(new DefaultFailureHandler());
+        // failure handler is added on a separate route with no name (the HTTP request root span name is taken from
+        // the last-applied named route, we don't want to overwrite the name of a specific route here)
+        matchAllFailuresRoute.failureHandler(ctx -> {
```

Comment:
is it by intention that we are creating two Route instances that match any URI, i.e. matchAllFailuresRoute and matchAllRoute? Or would it be sufficient to just create *matchAllRoute* and set this failure handler on it?